File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        int ans=0;
        unordered_map<int,int> umap;
        int n = tasks.size();
        for(int i=0; i<n; i++){
            umap[tasks[i]]++;
        }
        sort(begin(tasks), end(tasks));
        for(int i=0; i<n; ){
            int count = umap[tasks[i]];
            if(count>=2){
                if(count<=5 && count%2==0 && count%3!=0){
                    umap[tasks[i]] = count-2;
                    i+=2;
                } else{
                    umap[tasks[i]] = count-3;
                    i+=3;
                }
                ans++;
            } else {
                return -1;
            }
        }
        return ans;
    }
};

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.java
class Solution {
    public int minimumRounds(int[] tasks) {
        Arrays.sort(tasks);
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        int ans = 0;
        for(int i=0; i<tasks.length; i++)
            hmap.put(tasks[i], hmap.getOrDefault(tasks[i], 0) + 1);
        
        for(int i=0; i<tasks.length; ){
            int count = hmap.get(tasks[i]);
            if(count>=2){
                if(count<=5 && count%2==0 && count%3!=0){
                    hmap.put(tasks[i], count-2);
                    i += 2;
                } else{
                    hmap.put(tasks[i], count-3);
                    i += 3;
                }
                ans++;
            } else{
                return -1;
            }
        }
        return ans;
    }
}